rs/boundary_node/anonymization/backend/src/lib.rs

```diff
@@ -129,8 +141,84 @@ fn timers() {
                     ps.insert(p.to_owned(), ());
                 });
             });
+
+            // Remove stale public-keys
+            PUBLIC_KEYS.with(|ks| {
```

Comment:
A friendly suggestion, consider creating an abstraction layer for these operations, this would make the code more concise and remove low-level details of interaction with the storage variables. E.g. in `rate-limit` canister I created [this](https://github.com/dfinity/ic/blob/master/rs/boundary_node/rate_limits/canister/state.rs#L17). An additional benefit of abstraction is that You can test easier.